controllers/storagecluster/cephcluster.go

```diff
@@ -120,12 +133,6 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		}
 		monitoringIP := endpointR.Data["MonitoringEndpoint"]
 		monitoringPort := endpointR.Data["MonitoringPort"]
-		if err := verifyMonitoringEndpoints(monitoringIP, monitoringPort, r.Log); err != nil {
-			r.Log.Error(err, "Could not connect to the Monitoring Endpoints.",
-				"CephCluster", klog.KRef(sc.Namespace, sc.Name))
-			return reconcile.Result{}, err
-		}
-		r.Log.Info("Monitoring Information found. Monitoring will be enabled on the external cluster.", "CephCluster", klog.KRef(sc.Namespace, sc.Name))
 		cephCluster = newExternalCephCluster(sc, r.images.Ceph, monitoringIP, monitoringPort)
```

Comment:
Sending the `monitoringIP` to `newExternalCephCluster` will override the one that rook manages internally with the one from the initial config and this can cause problems. Up until now, the `verifyMonitoringEndpoints` prevented that from happening.

Reply:
At line#136, `newExternalCephCluster()` function will just initialize the variable `cephCluster`. This variable, `cephCluster` will only be used (to create the external cephcluster) when there is no existing CephCluster found (at line#193). Before create the CephCluster, we verify the endpoints (only in case of an external cluster mode).
So setting `monitoringIP` (and port) to `newExternalCephCluster()` function won't set or override the existing CephCluster spec.

Reply:
Yes, I think this is an issue. 
IIUC Before the change the cephcluster was reconciled every iteration while now we only reconcile it on creation - this is a major change in behavior. 

I think we should focus on doing the check only once, not to fully stop reconciliation after creation (as it can create other unrelated bugs)

Reply:
> Yes, I think this is an issue. IIUC Before the change the cephcluster was reconciled every iteration while now we only reconcile it on creation - this is a major change in behavior.

In the new change, we haven't touched any part of the previous code which affects `CephCluster`'s reconciliation. We just moved `verifyMonitoringEndpoints()` function down. No other change. Every other piece of code is as is.


> 
> I think we should focus on doing the check only once, not to fully stop reconciliation after creation (as it can create other unrelated bugs)

We do the check only once and we don't stop/alter the existing reconciling logic.

Reply:
Thanks @nb-ohad  for the clarification. I see that we are using the `cephCluster` object later (line#282 onwards) and use check the `Spec`. With this change, yes that will become a problem/bug.